File: src/backend/NHP_Domain/Entities/Utilisateur.cs
using System.Runtime.InteropServices.JavaScript;

namespace NHP_Domain.Entities;

public class Utilisateur
{
    public required String Id { get; set; }
    public required String MailUtilisateur { get; set; }
    public String? NomUtilisateur { get; set; }
    public String? PrenomUtilisateur { get; set; }
    public String? CiviliteUtilisateur { get; set; }
    public String? TelephoneUtilisateur { get; set; }
    public bool NewsLetterUtilisateur { get; set; }
    
    //ManyToOne
    //public virtual Role? Role { get; set; }
    public virtual Adresse? Adresse { get; set; }
    
    //OneToMany
    //public virtual List<Reservation> Reservation { get; set; }
    //public virtual List<Avis> Avis { get; set; }

    public override string ToString()
    {
        return $"mailUtilisateur: {MailUtilisateur} " +
               $"nomUtilisateur: {NomUtilisateur} " +
               $"prenomUtilisateur: {PrenomUtilisateur} " +
               $"civiliteUtilisateur: {CiviliteUtilisateur}" +
               $"telephoneUtilisateur: {TelephoneUtilisateur}" +
               $"newsLetterUtilisateur: {NewsLetterUtilisateur}" +
               //$"role: {Role}" +
               $"adresse: {Adresse}";
    }
}

File: src/frontend/src/domain/entities/Utilisateur.ts
import type { IAdresse } from '@/domain/entities/Adresse';
import type { IReservation } from '@/domain/entities/Reservation';
import type { IAvis } from '@/domain/entities/Avis';
import type { RoleType } from '@/domain/entities/Role';

export interface IUtilisateur {
    id: number;
    mailUtilisateur: string;
    nomUtilisateur?: string;
    prenomUtilisateur?: string;
    mdpUtilisateur?: string;
    civiliteUtilisateur?: string;
    telephoneUtilisateur?: string;
    newsLetterUtilisateur: boolean;
    role?: RoleType;
    adresse?: IAdresse;
    reservations?: IReservation[];
    avis?: IAvis[];
}

export class Utilisateur implements IUtilisateur {
    constructor(
        public id: number,
        public mailUtilisateur: string,
        public newsLetterUtilisateur: boolean,
        public nomUtilisateur?: string,
        public prenomUtilisateur?: string,
        public mdpUtilisateur?: string,
        public civiliteUtilisateur?: string,
        public telephoneUtilisateur?: string,
        public role?: RoleType,
        public adresse?: IAdresse,
        public reservations?: IReservation[],
        public avis?: IAvis[]
    ) {}
}